tests/test_notify_suppliers_whether_application_made_for_framework.py

```diff
@@ -0,0 +1,202 @@
+                "user 'KzsrnOhCq4tqbGFMsflgS7ig1QLRr0nFJrcrEIlOlbU='"
+            ),
+            mock.call(
+                "Sending 'application_made' email to supplier '712346' "
+                "user 'iYYo4oiQ-Te98Ak5He9Ch5xAGkvPG1_STnONn12oy7s='"
+            )
+        ]
+
+    def test_notify_suppliers_whether_application_made_dry_run(self):
+        assert notify_suppliers_whether_application_made(
+            self.data_api_client,
+            self.notify_client,
+            'g-cloud-12',
+            self.logger,
+            dry_run=True
+        ) == 0
+
+        assert self.notify_client.send_email.call_args_list == []
+        assert self.logger.info.call_args_list == [
+            mock.call("[Dry Run] Supplier '712345'"),
+            mock.call(
+                "[Dry Run] Sending 'application_not_made' email to supplier '712345' "
+                "user 's2qDcB8cMZHhlyLW-QJ0vBtVAf5p6_MzE-RA_ksP4hA='"
+            ),
+            mock.call("[Dry Run] Supplier '712346'"),
+            mock.call(
+                "[Dry Run] Sending 'application_made' email to supplier '712346' "
+                "user 'KzsrnOhCq4tqbGFMsflgS7ig1QLRr0nFJrcrEIlOlbU='"
+            ),
+            mock.call(
+                "[Dry Run] Sending 'application_made' email to supplier '712346' "
+                "user 'iYYo4oiQ-Te98Ak5He9Ch5xAGkvPG1_STnONn12oy7s='"
+            )
+        ]
+
+    def test_notify_suppliers_whether_application_made_supplier_ids_list(self):
+        assert notify_suppliers_whether_application_made(
+            self.data_api_client,
+            self.notify_client,
+            'g-cloud-12',
+            self.logger,
+            supplier_ids=[712345]
+        ) == 0
+
+        assert self.notify_client.send_email.call_args_list == [
+            mock.call(
+                "user1@example.com",
+                NOTIFY_TEMPLATES['application_not_made'],
+                {
+                    'intention_to_award_at': 'Sunday 31 December 2017',
+                    'framework_name': 'G-Cloud 12',
+                    'framework_slug': 'g-cloud-12',
+                    'applied': False
+                },
+                allow_resend=False
+            )
+        ]
+        assert self.logger.info.call_args_list == [
+            mock.call("Supplier '712345'"),
+            mock.call(
+                "Sending 'application_not_made' email to supplier '712345' "
+                "user 's2qDcB8cMZHhlyLW-QJ0vBtVAf5p6_MzE-RA_ksP4hA='"
+            )
+        ]
+
+    def test_notify_suppliers_whether_application_made_email_error_logs_supplier_id(self):
+        self.notify_client.send_email.side_effect = [
+            EmailError("Arghhh!"),  # The first user email fails
+            None,
+            None
+        ]
+
+        assert notify_suppliers_whether_application_made(
+            self.data_api_client,
+            self.notify_client,
+            'g-cloud-12',
+            self.logger
+        ) == 1
```

Comment:
Possibly my lack of context. But I am not sure of the meaning of `0` and `1` in these assertions. Is it possible to explain this at all?

Reply:
Yep, that's a good call - it's the expected number of errors returned by the function (fed into the exit code of the top level script). I'm happy to make this a named var in the test so it's clear - e.g `expected_error_count`.